compiler/common-artifacts/src/TestDataGenerator.cpp

```diff
@@ -176,6 +212,23 @@ int entry(int argc, char **argv)
   std::unique_ptr<H5::Group> output_value_group =
     std::make_unique<H5::Group>(output_file.createGroup("value"));
 
+  std::string range_name;
+  int32_t range_min = 0;
+  int32_t range_max = 0;
+  bool range_check = false;
+  bool range_input_found = false;
+  if (arser["--input_range"])
+  {
+    // NOTE limitation: we can only set one input range
+    // TODO expand this for multiple inputs
+    std::vector<std::string> values = arser.get<std::vector<std::string>>("--input_range");
+    assert(values.size() == 3);
+    range_name = values.at(0);
+    range_min = std::atoi(values.at(1).c_str());
```

Comment:
According to https://www.cplusplus.com/reference/cstdlib/atoi/, `atoi` never throws even the string is not a number format.
For example, `std::atoi("1a3")` in Ubuntu returns `1`.
So how about adding an assert like following?

```cpp
bool is_num(std::string num_string)
{
  for (auto ch : num_string)
    if (std::isdigit(ch) == false)
      return false;
  return true;
}

assert(is_num(values.at(1)));
assert(is_num(values.at(2)));
```

Reply:
@llFreetimell I think throwing an error or returning failure(-1) looks good as well because that kind of value like `1a3` could be given from user, which isn't a bit proper for `assert`.

Reply:
I'll leave as TODO